drivers/espi/espi_taf_npcx.c

```diff
@@ -423,6 +459,66 @@ static int espi_taf_npcx_flash_unsuccess(const struct device *dev, struct espi_s
 	return 0;
 }
 
+void espi_taf_work(struct k_work *item)
+{
+	struct espi_taf_npcx_data *info = CONTAINER_OF(item, struct espi_taf_npcx_data, work);
+	int ret = 0;
+
+	struct espi_taf_npcx_pckt taf_data;
+	struct espi_saf_packet pckt_taf;
+
+	pckt_taf.flash_addr = info->address;
+	pckt_taf.len = info->length;
+	taf_data.tag = info->taf_tag;
+	if (info->taf_type == NPCX_ESPI_TAF_REQ_WRITE) {
+		taf_data.data = (uint8_t *)info->src;
+	} else {
+		taf_data.data = NULL;
+	}
+	pckt_taf.buf = (uint8_t *)&taf_data;
+
+	switch (info->taf_type) {
+#if defined(CONFIG_ESPI_TAF_MANUAL_MODE)
+	case NPCX_ESPI_TAF_REQ_READ:
+		ret = espi_taf_npcx_flash_read(info->host_dev, &pckt_taf);
+		break;
+#endif
+	case NPCX_ESPI_TAF_REQ_ERASE:
+		ret = espi_taf_npcx_flash_erase(info->host_dev, &pckt_taf);
+		break;
+	case NPCX_ESPI_TAF_REQ_WRITE:
+		ret = espi_taf_npcx_flash_write(info->host_dev, &pckt_taf);
+		break;
+	}
+
+	if (ret != 0) {
+		ret = espi_taf_npcx_flash_unsuccess(info->host_dev, &pckt_taf);
+	}
+}
+
+static void espi_taf_event_handler(const struct device *dev, struct espi_callback *cb,
+				   struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_SAF_NOTIFICATION) {
+		if (event.evt_details == ESPI_CHANNEL_FLASH) {
```

Comment:
I would rewrite this as

```c
if (event.evt_type != ESPI_BUS_SAF_NOTIFICATION ||
    event.evt_details != ESPI_CHANNEL_FLASH) {
        return;
}
```

just to keep the normal execution inline

Reply:
Done.